src/PsfFlux.cc

```diff
@@ -110,10 +113,15 @@ void PsfFluxAlgorithm::measure(afw::table::SourceRecord& measRecord,
     // variance as if we had, so we'll apply the weights to the model now, and update alpha.
     result.instFluxErr = std::sqrt(model.array().square().matrix().dot(variance.cast<PsfPixel>())) / alpha;
     measRecord.set(_areaKey, model.sum() / alpha);
+    measRecord.set(_npixelsKey, fitRegion.getSpans()->getArea());
     if (!std::isfinite(result.instFlux) || !std::isfinite(result.instFluxErr)) {
         throw LSST_EXCEPT(PixelValueError, "Invalid pixel value detected in image.");
     }
     measRecord.set(_instFluxResultKey, result);
+    auto chi2 = ((data.cast<PsfPixel>() - result.instFlux * model).array().square() *
+                 variance.cast<PsfPixel>().array().rsqrt())
+                        .sum();
+    measRecord.set(_chi2Key, chi2);
```

Comment:
While I'm not a particular fan of reduced χ^2 (AFAIK statisticians don't take it or "degrees of freedom" seriously), I don't see where the 1/N factor you promised comes from?

Reply:
The 1/N is a lie, and I've removed it from the comments/docs.